File: Selectionsort.c

#include <stdio.h>    
        
void swap(int *A, int i, int j) {    
          int temp = A[i];    
          A[i] = A[j];    
          A[j] = temp;    
}    
        
int findMinIndex(int *A, int start, int end) {    
     int min_index = start;    
        
     ++start;    
        
     while(start < end) {    
         if(A[start] < A[min_index])    
             min_index = start;    
        
         ++start;    
     }    
        
     return min_index;    
}    
        
void selectionSort(int *A, int n) {    
    for(int i = 0; i < n - 1; ++i) {    
        int min_index = findMinIndex(A, i, n);    
        
        if(i != min_index)    
            swap(A, i, min_index);    
    }    
}    
        
int main() {    
    int A[] = {5, 2, 6, 7, 2, 1, 0, 3}, n = 8;    
        
    selectionSort(A, n);    
        
    for(int i = 0; i < n; ++i)    
        printf("%d ", A[i]);  
    return 0;    
}

File: SelectionSort.py
# -*- coding: utf-8 -*-
"""
 python code to implement Selection Sort.

@author: rajmadhu0406
"""

def swap(List, x, y):
    temp = List[x]
    List[x] = List[y]
    List[y] = temp

def SelectionSort(List):
    min_index = 0
    n = len(List)
    for i in range(n):
        min_index = i
        for j in range(i,n):
            if(List[j] < List[min_index]):
                min_index = j
        swap(List, i, min_index)
    #print(List)
    return List

List = [4, 2, 5, 1, 3]
print("Sorted List : ", SelectionSort(List))
    
            
        
    